app/domain/authentication/o_auth/verify_and_decode_token.rb

```diff
@@ -1,26 +1,31 @@
 require 'jwt'
 
 module Authentication
-  module AuthnOidc
+  module OAuth
 
-    Err = Errors::Authentication::AuthnOidc
-    Log = LogMessages::Authentication::AuthnOidc
+    Err = Errors::Authentication::OAuth
+    Log = LogMessages::Authentication::OAuth
     # Possible Errors Raised:
     # TokenExpired, TokenDecodeFailed, TokenVerifyFailed
 
+    # This class decodes and verifies JWT tokens, issued by an OAuthn 2.0 Identity Provider.
+    # It first retrieves the JWKs from the identity provider and sets them as verification options
+    # for Authentication::Jwt::VerifyAndDecodeToken. That class does the offline verification and decode,
+    # using the JWKs retrieved from the OAuthn 2.0 Identity Provider.
     VerifyAndDecodeToken = CommandClass.new(
       dependencies: {
         # We have a ConcurrencyLimitedCache which wraps a RateLimitedCache which wraps a FetchProviderCertificate class
         fetch_provider_certificate: ::Util::ConcurrencyLimitedCache.new(
           ::Util::RateLimitedCache.new(
-            ::Authentication::AuthnOidc::FetchProviderCertificates.new,
+            ::Authentication::OAuth::FetchProviderCertificates.new,
             refreshes_per_interval: 10,
             rate_limit_interval:    300, # 300 seconds (every 5 mins)
             logger: Rails.logger
           ),
           max_concurrent_requests: 3, # TODO: Should be dynamic calculation
           logger: Rails.logger
         ),
+        verify_and_decode_token: ::Authentication::Jwt::VerifyAndDecodeToken.new,
```

Comment:
What is the difference between jwt and o_auth/verify_decode_token. It is confusing that both have the same naming

Reply:
>  It first retrieves the JWKs from the identity provider and sets them as verification options for Authentication::Jwt::VerifyAndDecodeToken. That class does the offline verification and decode, using the JWKs retrieved from the OAuthn 2.0 Identity Provider.

Would like to discuss this comment because the last sentence isnt too clear to me

Reply:
That's why I added a comment on each class. One is OAuth scoped and one is JWT scoped and they both verify and decode a token. If you have a better suggestion for naming tell me.

Reply:
what's unclear?